src/overlay005/field_control.c

```diff
@@ -0,0 +1,1092 @@
+
+        sub_0202CF28(v2, (1 + 10));
+        sub_0202CFEC(v2, 15);
+        sub_0203E880(fieldSystem, 2031, NULL);
+
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static BOOL Field_UpdateVsSeeker (FieldSystem *fieldSystem)
+{
+    ov5_021DBB94(fieldSystem);
+    return FALSE;
+}
+
+static BOOL Field_UpdatePokeRadar (FieldSystem *fieldSystem)
+{
+    RadarChargeStep(fieldSystem);
+    return FALSE;
+}
+
+static BOOL Field_UpdateRepel (FieldSystem *fieldSystem)
+{
+    return ov6_02246BF4(fieldSystem->saveData, fieldSystem);
+}
+
+static BOOL Field_UpdateFriendship (FieldSystem *fieldSystem)
+{
+    UnkStruct_020507E4 *events;
+    BOOL ret = FALSE;
+
+    events = SaveData_Events(fieldSystem->saveData);
+    u16 steps = sub_0206B44C(events);
+
+    steps++;
+
+    if (steps >= 128) {
+        steps = 0;
+        ret = TRUE;
+    }
+
+    sub_0206B45C(events, steps);
+
+    return ret;
+}
+
+static void Field_CalculateFriendship (FieldSystem *fieldSystem)
+{
+    int i, partyCount;
+    Pokemon *mon;
+    Party *party = Party_GetFromSavedata(fieldSystem->saveData);
+    u16 mapID = MapHeader_GetMapLabelTextID(fieldSystem->unk_1C->unk_00);
+
+    partyCount = Party_GetCurrentCount(party);
+
+    for (i = 0; i < partyCount; i++) {
+        mon = Party_GetPokemonBySlotIndex(party, i);
+        Pokemon_UpdateFriendship(mon, 5, mapID);
+    }
+}
+
+static BOOL Field_UpdatePoison (FieldSystem *fieldSystem)
+{
+    Party *party = Party_GetFromSavedata(fieldSystem->saveData);
+    u16 *poisonSteps = sub_0203A78C(sub_0203A790(fieldSystem->saveData));
+
+    (*poisonSteps)++;
+    (*poisonSteps) %= 4;
+
+    if (*poisonSteps != 0) {
+        return FALSE;
+    }
+
+    switch (sub_02054B04(party, MapHeader_GetMapLabelTextID(fieldSystem->unk_1C->unk_00))) {
+    case 0:
+        return FALSE;
+    case 1:
+        ov5_021EF518(fieldSystem->unk_04->unk_20);
+        return FALSE;
+    case 2:
+        ov5_021EF518(fieldSystem->unk_04->unk_20);
+        sub_0203E880(fieldSystem, 2003, NULL);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static BOOL Field_UpdateSafari (FieldSystem *fieldSystem)
+{
+    if (sub_0206AE5C(SaveData_Events(fieldSystem->saveData)) == FALSE) {
+        return FALSE;
+    }
+
+    u16 *balls = sub_0203A784(sub_0203A790(fieldSystem->saveData));
+
+    if (*balls == 0) {
+        sub_0203E880(fieldSystem, 8802, NULL);
+        return TRUE;
+    }
+
+    u16 *steps = sub_0203A788(sub_0203A790(fieldSystem->saveData));
+    (*steps)++;
+
+    if (*steps >= 500) {
+        sub_0203E880(fieldSystem, 8801, NULL);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+static void Field_PlayerPos (const FieldSystem *fieldSystem, int *playerX, int *playerZ)
+{
+    *playerX = Player_XPos(fieldSystem->playerAvatar);
+    *playerZ = Player_ZPos(fieldSystem->playerAvatar);
+}
+
+static void Field_NextPlayerPos (const FieldSystem *fieldSystem, int *playerX, int *playerZ)
```

Comment:
```suggestion
static void Field_MoveOneStep (const FieldSystem *fieldSystem, int *playerX, int *playerZ)
```

Reply:
Maybe even just `Field_Step` (and others accordingly)?